### PR TITLE
pkp/pkp-lib#2500: Modify NativeXMLSubmissionFileFilter::handleRevisionChildElement() to consistently return a temporary filename.

### PR DESCRIPTION
**Pull request by ctgraham**
Shared library component of #2500.

### CODE REVIEW

**Comment by asmecher** [comment]
@bozana, would you mind taking a look at this one?

File: plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php [hunk 1]
```diff
@@ -160,6 +160,9 @@ function handleRevisionElement($node, $stageId, $fileId) {
 				$filename = $this->handleRevisionChildElement($n, $submission, $submissionFile);
 			}
 		}
+		if (!$filename) {
```
Comment (bozana) [inline plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php]: Isn't it that the $filename will always be returned?

File: plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php [hunk 2]
```diff
@@ -276,8 +279,32 @@ function handleRevisionChildElement($node, $submission, $submissionFile) {
...
+					import('lib.pkp.classes.file.FileWrapper');
+					$wrapper = FileWrapper::wrapper($filesrc);
+					file_put_contents($temporaryFilename, $wrapper->contents());
+					if (!filesize($temporaryFilename)) {
+						$deployment->addError(ASSOC_TYPE_SUBMISSION, $submission->getId(), __('plugins.importexport.common.export.error.inputFileNotReadable', array('param' => $filesrc)));
+					}
+					return $temporaryFilename;
+				}
+				// local file (absolute path)
+				if (substr($filesrc, 1, 1) === '/') {
```
Comment (bozana) [inline plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php]: Should here maybe also be checked `if` `is_readable` (like in the next if statement?

File: plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php [hunk 3]
```diff
@@ -276,8 +279,32 @@ function handleRevisionChildElement($node, $submission, $submissionFile) {
...
+					copy($filesrc, $temporaryFilename);
```
Comment (bozana) [inline plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php]: Could `copy` here and below fail?

File: plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php [hunk 4]
```diff
@@ -276,8 +279,32 @@ function handleRevisionChildElement($node, $submission, $submissionFile) {
...
+				$temporaryFilename = tempnam($temporaryFileManager->getBasePath(), 'src');
 				// Allow wrappers to handle URLs
-				return $node->getAttribute('src');
+				$filesrc = $node->getAttribute('src');
+				if (preg_match('|\w+://.+|', $filesrc)) {
+					import('lib.pkp.classes.file.FileWrapper');
+					$wrapper = FileWrapper::wrapper($filesrc);
+					file_put_contents($temporaryFilename, $wrapper->contents());
+					if (!filesize($temporaryFilename)) {
+						$deployment->addError(ASSOC_TYPE_SUBMISSION, $submission->getId(), __('plugins.importexport.common.export.error.inputFileNotReadable', array('param' => $filesrc)));
```
Comment (bozana) [inline plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php]: This locale key does not exist yet, right? Would you mind inserting it?

Comment (bozana) [inline plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php]: Oh, I see it in another PR... Sorry! :-\

File: plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php (hunk 1 above)
Comment (ctgraham) [inline plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php]: Yes, switching this to a file_exists() and filesize().

File: plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php (hunk 4 above)
Comment (ctgraham) [inline plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php]: This ultimately deserves a new key anyway.

File: plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php (hunk 3 above)
Comment (ctgraham) [inline plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php]: Yes, refactoring.

**Comment by ctgraham** [comment]
@bozana, I've refactored this commit based on your comments.

**Pull request merged**